packages/legend-application-studio/src/stores/editor-state/element-editor-state/service/ServiceExecutionState.ts

```diff
@@ -656,21 +678,19 @@ export abstract class ServicePureExecutionState extends ServiceExecutionState {
   }
 
   useCustomRuntime(): void {
-    if (this.selectedExecutionContextState) {
-      const customRuntime = new EngineRuntime();
-      runtime_addMapping(
-        customRuntime,
-        PackageableElementExplicitReference.create(
-          this.selectedExecutionContextState.executionContext.mapping.value,
-        ),
-      );
-      decorateRuntimeWithNewMapping(
-        this.selectedExecutionContextState.executionContext.runtime,
-        this.selectedExecutionContextState.executionContext.mapping.value,
-        this.editorStore,
-      );
-      this.selectedExecutionContextState.setRuntime(customRuntime);
-    }
+    const customRuntime = new EngineRuntime();
+    runtime_addMapping(
+      customRuntime,
+      PackageableElementExplicitReference.create(
+        this.selectedExecutionContextState!.executionContext.mapping.value,
```

Comment:
lets use guaranteeNonUllable for these two. 
guaranteeNonNullable(this.selectedExecutionContextState?.executionContext.mapping.value)

Reply:
same her as well